idx/memory/partitioned_idx.go

```diff
@@ -544,3 +544,39 @@ func (p *PartitionedMemoryIdx) idsByTagQuery(orgId uint32, query TagQuery) IdSet
 	}
 	return response
 }
+
+func (p *PartitionedMemoryIdx) MetaTagRecordList(orgId uint32) []idx.MetaTagRecord {
+	for _, m := range p.Partition {
+		// all partitions should have all meta records
+		return m.MetaTagRecordList(orgId)
+	}
+	return nil
+}
+
+func (p *PartitionedMemoryIdx) MetaTagRecordUpsert(orgId uint32, rawRecord idx.MetaTagRecord) (idx.MetaTagRecord, bool, error) {
+	g, _ := errgroup.WithContext(context.Background())
+
+	var i int
+	var record idx.MetaTagRecord
+	var created bool
+	for _, m := range p.Partition {
+		g.Go(func() error {
+			var err error
+			if i == 0 {
+				record, created, err = m.MetaTagRecordUpsert(orgId, rawRecord)
+			} else {
+				_, _, err = m.MetaTagRecordUpsert(orgId, rawRecord)
+			}
```

Comment:
you are not closing over the `m` variable correctly.
run this and then try without the `m := m` line
```
package main

import (
	"context"
	"fmt"
	"time"

	"golang.org/x/sync/errgroup"
)

func main() {
	g, _ := errgroup.WithContext(context.Background())
	parts := []int{0, 1, 2, 3, 4, 5}
	for _, m := range parts {
		m := m
		g.Go(func() error {
			fmt.Println(m)
			return nil
		})
	}
	time.Sleep(time.Second)
}
```

Reply:
good catch